suite/tests/linux/sigtrans.c

```diff
@@ -0,0 +1,154 @@
+/* **********************************************************
+ * Copyright (c) 2025 Arm Limited. All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of ARM Limited nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL ARM LIMITED OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/* Target a block of instructions that use the stolen register with a
+ * synchronous signal to check that the app state is correctly
+ * recreated and the instruction is not run a second time after
+ * returning from the signal handler.
+ */
+
+#include "tools.h"
+
+#include <signal.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <time.h>
+
+/* The signal handler will copy src to dst. */
+static volatile int volatile_src;
+static volatile int volatile_dst;
+
+void
+handler(int signum, siginfo_t *info, void *ucontext)
+{
+    volatile_dst = volatile_src;
+}
+
+void
+fail(const char *s)
+{
+    perror(s);
+    exit(1);
+}
+
+/* XXX: These initial values must be large enough to avoid i#7675, but
+ * should probably be reduced when that bug is fixed.
+ */
+int iters = 1000;
+unsigned long init_param = 10000;
+
+bool
+test_code(int *adjust)
+{
+    unsigned long count = 0;
+    for (int i = 0; i < iters * 2; i++) {
```

Comment:
Clearer to set iters to 2000 and use plain iters here. This is the main use, so you expect the var to be the actual iter count.

Reply:
That's not just aesthetics, either, because I wasn't checking for overflow when doubling `iters` there!

Done.